### PR TITLE
Feat/valuemap

### PR DESCRIPTION
**Pull request by wanoz**
Fixed some bugs and added max_reward_spread argument.
Might have an issue regarding x,y orientation.

### CODE REVIEW

File: python3/totoro_agent/utils/util_functions.py [hunk 1]
```diff
@@ -363,6 +362,59 @@ def is_movement(action):
...
+    else:
+        # Reward assigned is 0.
+        pass
+
+    # Remove map padding
+    value_map = value_map[world_dim[0]-1:world_dim[0]+pad_dim[0],world_dim[0]-1:world_dim[0]+pad_dim[0]]
+
+    return value_map
 
 def get_value_map(world, walls, game_objects, reward_map, pinch_points=None, use_default=True):
```
Comment (chrisrabe) [inline python3/totoro_agent/utils/util_functions.py:419]: as discussed, need to update offset and orientation

**Review by chrisrabe — CHANGES_REQUESTED** [review]
As discussed, need to update orientation of value map

File: python3/totoro_agent/utils/util_functions.py [hunk 2]
```diff
@@ -363,6 +362,59 @@ def is_movement(action):
...
+    Returns a numpy array representing the updated reward value map.
+    '''
+    # Add map padding
+    pad_dim = (world_dim[0]-1,world_dim[1]-1)
+    value_map = np.pad(value_map, (pad_dim,pad_dim), 'constant', constant_values=OUTER_MAP_VALUES)
+    rval_offset = pad_dim[0]-1
+
+    reward = rval[2]
+    reward_discount = reward/abs(reward)
+    rval_offset = 8 # padding offset
```
[the remark this answers is not in the document]
Comment (wanoz) [reply python3/totoro_agent/utils/util_functions.py]: Done

File: python3/totoro_agent/utils/util_functions.py [hunk 3]
```diff
@@ -389,13 +441,18 @@ def get_value_map(world, walls, game_objects, reward_map, pinch_points=None, use
     # TODO are there numpy helper functions to help with these logic?
 
     # create 2D matrix filled with zeroes
-    value_map = np.zeros(get_world_dimension(world))
+    world_dim = get_world_dimension(world)
+    value_map = np.zeros(world_dim)
+    reward_entity = []
```
[the remark this answers is not in the document]
Comment (wanoz) [reply python3/totoro_agent/utils/util_functions.py:447]: Which variable is referring to here?

File: python3/totoro_agent/utils/util_functions.py (hunk 1 above)
Comment (wanoz) [reply python3/totoro_agent/utils/util_functions.py:420]: Updated as discussed.

**Pull request closed without merging**